File: frontend/src/types/category.ts
export interface Category {
  id: string;
  name: string;
  slug: string;
  description?: string;
  displayOrder?: number;
  isActive?: boolean;
  createdAt: string;
}

export interface CategoryFormData {
  name: string;
  slug: string;
  description?: string;
  displayOrder?: number;
  isActive?: boolean;
}

File: backend/src/ProductManagement.Domain/Entities/Category.cs
using System;
using System.Collections.Generic;

namespace ProductManagement.Domain.Entities;

public partial class Category
{
    public Guid Id { get; set; }

    public string Name { get; set; } = null!;

    public string Slug { get; set; } = null!;

    public string? Description { get; set; }

    public int? DisplayOrder { get; set; }

    public bool? IsActive { get; set; }

    public DateTime CreatedAt { get; set; }

    public virtual ICollection<Product> Products { get; set; } = new List<Product>();
}
